xbmc/utils/ArtUtils.cpp

```diff
@@ -28,6 +31,126 @@ using namespace XFILE;
 namespace KODI::ART
 {
 
+void FillInDefaultIcon(CFileItem& item)
+{
+  if (URIUtils::IsPVRGuideItem(item.GetPath()))
+  {
+    // epg items never have a default icon. no need to execute this expensive method.
+    // when filling epg grid window, easily tens of thousands of epg items are processed.
+    return;
+  }
+
+  //CLog::Log(LOGINFO, "FillInDefaultIcon({})", pItem->GetLabel());
+  // find the default icon for a file or folder item
+  // for files this can be the (depending on the file type)
+  //   default picture for photo's
+  //   default picture for songs
+  //   default picture for videos
+  //   default picture for shortcuts
+  //   default picture for playlists
+  //
+  // for folders
+  //   for .. folders the default picture for parent folder
+  //   for other folders the defaultFolder.png
+
+  if (item.GetArt("icon").empty())
+  {
+    if (!item.m_bIsFolder)
+    {
+      /* To reduce the average runtime of this code, this list should
+       * be ordered with most frequently seen types first.  Also bear
+       * in mind the complexity of the code behind the check in the
+       * case of IsWhatever() returns false.
+       */
+      if (item.IsPVRChannel())
+      {
+        if (item.GetPVRChannelInfoTag()->IsRadio())
```

Comment:
Seems this is the only reason for including "PVRChannel.h" (compile time dep to PVR component). Can you ple ase put a `@todo get rid of PVR compile time dependency` and I will pick this up later?

Reply:
Or if you like, you could introduce `bool CFileItem::IsPVRRadioChannel() const`?

Reply:
added the todo.